packages/Main/src/Renderer/LayeredMaterial.ts

```diff
@@ -94,86 +94,35 @@ const rtCache = new LRUCache<string, THREE.WebGLArrayRenderTarget>({
-    const uTextures = uniforms.textures;
-    const uOffsetScales = uniforms.offsetScales.value;
-    const uTextureCount = uniforms.textureCount;
-
-    // Flatten the 2d array: [i, j] -> layers[_layerIds[i]].textures[j]
-    let count = 0;
-    let width = 0;
-    let height = 0;
-
-    // Determine total count of textures and dimensions
-    // (assuming all textures are same size)
-    let textureSetId: string = type;
-    for (const tile of tiles) {
-        // FIXME: RasterElevationTile are always passed to this function alone
-        // so this works, but it's really not great even ignoring the dynamic
-        // addition of a field.
-        // @ts-expect-error: adding field to passed layer
-        tile.textureOffset = count;
-
-        for (
-            let i = 0;
-            i < tile.textures.length && count < max;
-            ++i, ++count
-        ) {
-            const texture = tile.textures[i];
-            if (!texture) { continue; }
-
-            textureSetId += `${texture.id}.`;
-            uOffsetScales[count] = tile.offsetScales[i];
-            uLayers[count] = tile;
-
-            const img = texture.image;
-            if (!img || img.width <= 0 || img.height <= 0) {
-                console.error('Texture image not loaded or has zero dimensions');
-                uTextureCount.value = 0;
-                return;
-            } else if (count == 0) {
-                width = img.width;
-                height = img.height;
-            } else if (width !== img.width || height !== img.height) {
-                console.error('Texture dimensions mismatch');
-                uTextureCount.value = 0;
-                return;
-            }
-        }
-    }
-
-    const cachedRT = rtCache.get(textureSetId);
-    if (cachedRT) {
-        uTextures.value = cachedRT.texture;
-        uTextureCount.value = count;
+    renderer: THREE.WebGLRenderer,
+    extent: Extent) {
+    if (tiles.find(tile => tile.textures.length == 0)) {
         return;
     }
 
-    // Memory management of these textures is only done by `textureArraysCache`,
-    // so we don't have to dispose of them manually.
-    const rt = makeDataArrayRenderTarget(width, height, count, tiles, max, renderer);
-    if (!rt) {
-        uTextureCount.value = 0;
-        return;
-    }
+    const textureSetId = extent.toString();
 
-    rtCache.set(textureSetId, rt);
-    uniforms.textures.value = rt.texture;
+    let renderTarget = rtCache.get(textureSetId);
 
-    if (count > max) {
-        console.warn(
-            `LayeredMaterial: Not enough texture units (${max} < ${count}), `
-            + 'excess textures have been discarded.',
-        );
+    if (!renderTarget) {
+        if (tiles.length == 0) {
+            return;
+        }
+
+        renderTarget = new THREE.WebGLRenderTarget(256, 256, { depthBuffer: false });
+
+        drawMap(renderTarget, tiles, renderer, extent);
+
+        rtCache.set(textureSetId, renderTarget);
```

Comment:
I was caching array render targets instead of array textures because it was impossible to extract the textures from the RTs and manage their memory separately, but maybe now that you're using simple render targets, this becomes possible and we can just maintain a cache of textures directly, and only use one RT for everything?